File: QuickSeries/app/src/main/java/com/test/quickseries/category/adapter/CategoryViewHolder.java
package com.test.quickseries.category.adapter;

import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.test.quickseries.databinding.CategoryItemBinding;

/**
 * Created by elton on 2017-10-15.
 */

public class CategoryViewHolder extends RecyclerView.ViewHolder {

    CategoryItemBinding binding;

    public CategoryViewHolder(View itemView) {
        super(itemView);
        binding = DataBindingUtil.bind(itemView);
    }
}

File: MobileTest/app/src/main/java/com/quickseries/mobiletest/adapters/category/CategoryViewHolder.kt
package com.quickseries.mobiletest.adapters.category

import android.support.v7.widget.AppCompatImageView
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import com.quickseries.mobiletest.R
import com.quickseries.mobiletestcore.models.CategoryModelBase


class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private lateinit var category: CategoryModelBase
    private val titleTextView = itemView.findViewById<TextView>(R.id.titleTextView)
    private val infoImageView = itemView.findViewById<AppCompatImageView>(R.id.infoImageView)

    var infoImageViewClick: CategoryClickDelegate? = null
    var itemClick: CategoryClickDelegate? = null

    init {

        infoImageView.isClickable = true
        infoImageView.setOnClickListener {
            infoImageViewClick?.invoke(category)
        }

        itemView.isClickable = true
        itemView.setOnClickListener {
            itemClick?.invoke(category)
        }
    }

    fun dataBind(category: CategoryModelBase) {
        this.category = category
        this.titleTextView.text = category.title
    }
}